packages/flutter/test/widgets/widget_inspector_test.dart

```diff
@@ -2216,6 +2223,97 @@ class _TestWidgetInspectorService extends TestWidgetInspectorService {
+      const String group = 'test-group';
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: Stack(
+            children: const <Widget>[
+              Text('a', textDirection: TextDirection.ltr),
+              Text('b', textDirection: TextDirection.ltr),
+              Text('c', textDirection: TextDirection.ltr),
+            ],
+          ),
+        ),
+      );
+      final Element elementA = find.text('a').evaluate().first;
+
+      service
+        ..disposeAllGroups()
+        ..resetPubRootDirectories()
+        ..setSelection(elementA, 'my-group');
+  
+      final Map<String, dynamic> jsonA = (await service.testExtension(
+        WidgetInspectorServiceExtensions.getSelectedWidget.name,
+        <String, String>{'objectGroup': 'my-group'},
+      ))! as Map<String, dynamic>;
+
+
+      final Map<String, Object?> creationLocation = jsonA['creationLocation']! as Map<String, Object?>;
+      expect(creationLocation, isNotNull);
+      final String testFile = creationLocation['file']! as String;
+      expect(testFile, endsWith('widget_inspector_test.dart'));
+      final List<String> segments = Uri.parse(testFile).pathSegments;
+      // Strip a couple subdirectories away to generate a plausible pub root
+      // directory.
+      final String pubRootTest = '/${segments.take(segments.length - 2).join('/')}';
+      service
+        ..resetPubRootDirectories()
+        ..addPubRootDirectories(<String>[pubRootTest]);
+
+      final Map<String, Object?> rootJson = (await service.testExtension(
+        WidgetInspectorServiceExtensions.getRootWidgetSummaryTreeWithPreviews.name,
+        <String, String>{'groupName': group},
+      ))! as Map<String, Object?>;
+      List<Object?> childrenJson = rootJson['children']! as List<Object?>;
+      // The tree of nodes returned contains all widgets created directly by the
+      // test.
+      childrenJson = rootJson['children']! as List<Object?>;
+      expect(childrenJson.length, equals(1));
+
+      List<Object?> alternateChildrenJson = (await service.testExtension(
+        WidgetInspectorServiceExtensions.getChildrenSummaryTree.name,
+        <String, String>{'arg': rootJson['objectId']! as String, 'objectGroup': group},
+      ))! as List<Object?>;
+      expect(alternateChildrenJson.length, equals(1));
+      Map<String, Object?> childJson = childrenJson[0]! as Map<String, Object?>;
+      Map<String, Object?> alternateChildJson = alternateChildrenJson[0]! as Map<String, Object?>;
+      expect(childJson['description'], startsWith('Directionality'));
+      expect(alternateChildJson['description'], startsWith('Directionality'));
+      expect(alternateChildJson['valueId'], equals(childJson['valueId']));
+
+      childrenJson = childJson['children']! as List<Object?>;
+      alternateChildrenJson = (await service.testExtension(
```

Comment:
This is a very long test that seems to test seperate testExtension calls.
Would it be possible to split this up into multiple tests to help make debugging easier if these ever end up failing?

Reply:
I copied this test from 'ext.flutter.inspector.getRootWidgetSummaryTree' above, and stripped out what I could. We need to call 'getChildrenSummaryTree' to access the summary tree for the children. Unfortunately the child with the 'textPreview' occurs a few levels down in the tree. I could try removing the Stack widget that is being pumped, but that would only eliminate one level of the tree.